### ISSUE TITLE
Re-introduce the concept of Entry and EntryKey

### ISSUE THREAD

**Issue by codeboten** [open-telemetry/opentelemetry-specification#442]
The introduction of the Correlations API simplified the API spec by removing the existing definition of an Entry and EntryKey. As per the discussion in #420, there may be a use-case to support an alternative to strings as identifiers in the Correlations API. Creating this issue to track the discussion to define what an Entry and an EntryKey should be, to unblock #420.

**Comment by carlosalberto** [comment]
Nice. I'd like to see all these things being properly addressed, and I'd rather have this part removed from the Spec for now.

**Comment by jmacd** [comment]
I would like to register this statement: https://github.com/open-telemetry/opentelemetry-specification/pull/420#issuecomment-582231334

**Comment by tigrannajaryan** [comment]
Do we still believe this is a must-have for GA?

**Comment by Oberon00** [comment]
As no one has spoken up, I'll move this to after-ga.

**Comment by ocelotl** [comment]
I asked Claude to review this issue and Claude suggests it be closed with this explanation:

> Discussion fizzled: Oberon00 moved it to after-ga (2020-09-15) noting 'no one has spoken up.' The proposal for non-string Entry/EntryKey abstraction was never revived, and the finalized baggage spec (baggage/api.md) settled firmly on string names and string values ('Baggage values are any valid UTF-8 strings'), making the re-introduction request obsolete against the GA design. Cross-ref #1377 is unrelated (an accidental reference).

To guard against a mistaken suggestion, Claude then independently re-verified this conclusion:

> Closeable but the category is stale-abandoned, not obsolete: this is a still-actionable enhancement (non-string Entry/EntryKey) that was deferred to release:after-ga on 2020-09-15 with "no one has spoken up," stripped of required-for-ga/p2, and never revived in ~5 years. GA baggage/api.md settling on string-only values is the status quo the request wanted to change (not a resolution), so "obsolete" is wrong. #1377 cross-ref is unrelated (java revert); the 2025 "reference" is an accidental fork commit collision ("should->SHOULD (#442)"). Issue still OPEN.

Claude evaluated this issue with medium confidence that it can be closed.

Claude has classified this issue as stale.

The intention of this comment is to make it easier for a human being to review this issue before making a decision.

**Comment by svrnm** [comment]
@codeboten can you take a look at @ocelotl's comment and close the issue if this is the right move, or update it with more recent details.
